FT013 - feat: added section-card component

damiankubecki opened the pull request:
This commit adds section card component

Commit:
feat: added section-card component

This commit adds section card component

Commit:
feat: updated section-card component

This commit updates section-card component

Commit:
feat: updated section-card component

This commit updates section-card component

EMEZYK commented on src/app/core/theme.component.ts:
```diff
@@ -1,8 +1,53 @@
+      console.log('modification...');
+    },
+    handleActivation: function () {
+      console.log('activation...');
```
zbędne console.log możemy usunąć

damiankubecki replied:
Te console.log'i znajdują się w theme componencie i mają na celu lepiej zobrazować działanie komponentu dla osoby, która w przyszłości będzie chciała z niego korzystać. Wydaje mi się, że w tym miejscu mogą się legalnie pojawić, bo być może jeszcze nie każdy ogarnia działanie outputów - tak jak ja dzień wcześniej 😎

EMEZYK commented on src/app/features/section/shared/section-card/section-card.component.scss:
```diff
@@ -0,0 +1,80 @@
+      }
+    }
+  }
+}
```
do ustalenia z grupą - czy przechodzimy na konwencję BEM

EMEZYK commented on src/app/features/section/shared/section-card/section-card.component.ts:
```diff
@@ -0,0 +1,32 @@
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SectionCardComponent implements OnInit {
+  @Input() name!: string;
+  @Input() numberOfMembers!: number;
+  @Input() isActive!: boolean;
```
co powiesz na to, żeby zamiast używania tutaj ! , podać wartości domyślne?  / ew. skonfigurować ts.config.json:"compilerOptions": {
  ///
  ,
  "strictPropertyInitialization":false
}  ?

damiankubecki replied:
Hmm wydaje mi się, że nie ma potrzeby przypisywać tym polom wartości domyślnych, ponieważ są wymagane, więc i tak zostaną przekazane. Moim zdaniem ten `!` wyróżnia inputy, które są `required` i nie mamy potrzeby patrzenia na selektor

EMEZYK replied:
ok, przekonales mnie kolego

EMEZYK commented on src/app/features/section/shared/section-card/section-card.component.scss:
```diff
@@ -0,0 +1,80 @@
+.card {
+  width: 270px;
+  border-radius: 5px;
+  background-color: var(--surface);
+  box-shadow: var(--box-shadow);
+  cursor: pointer;
+
```
a moze w sumie skorzystamy tutaj z materiala? https://material.angular.io/components/card/overview

damiankubecki committed:
feat: updated section-card component

This commit changes classes of html tags (removed BEM) and adds mat-card tag from Material

damiankubecki committed:
Merge branch 'develop' into FT013-feat-stworzenie-elementu-listy-sekcji

damiankubecki committed:
fix: removed unused type

This commit removes the unused type and resolves the names dupliaction conflict
